### PR TITLE
Mark user events

### PR DESCRIPTION
**Pull request by jtkpiotr**
@adamzima @pbzymek @s1mplex @lkudyba @mhib @quirk0o

### CODE REVIEW

File: app/models/google_event.rb [hunk 1]
```diff
@@ -14,15 +14,19 @@ class InvalidParamsError < StandardError
 
   class << self
     # You can specify custom fields: https://developers.google.com/google-apps/calendar/v3/reference/events
-    FIELDS = 'items(id, start, end, summary, recurrence, creator)'.freeze
+
+    def listing_options(fields, starting, ending)
+      {fields: fields, single_events: true, time_min: starting.rfc3339(9),
+       time_max: ending.rfc3339(9), time_zone: 'Europe/Warsaw'}
```
Comment (quirk0o) [inline app/models/google_event.rb]: Shouldn't we use the env variable for timezone?

Comment (mhib) [inline app/models/google_event.rb]: Good catch

File: app/models/google_event.rb [hunk 2]
```diff
@@ -14,27 +14,52 @@ class InvalidParamsError < StandardError
...
+    def merge_events(new_events, room, all_events)
+      return unless new_events
+      new_events.items&.each do |event|
+        normalize_event_datetime(event)
+        all_events[event.start.date_time.wday] << event.to_h.merge(conference_room: room)
+      end
+    end
+
+    def normalize_event_datetime(event)
+      event.start.date_time = EventGrouper.new_time_low event.start.date_time
```
Comment (pbzymek) [inline app/models/google_event.rb]: missing parenthesis on methods call

File: app/controllers/authentication_controller.rb [hunk 3]
```diff
@@ -1,9 +1,17 @@
 class AuthenticationController < ApplicationController
   def authenticate
-    session[:credentials] = GoogleOauth.get_user_credentials(params[:code])
+    set_session
     redirect_to root_path
   rescue
+    session.delete(:email)
     session.delete(:credentials)
```
Comment (quirk0o) [inline app/controllers/authentication_controller.rb:7]: Why not have a corresponding method to `set_session` e.g. `clear_session`

File: app/controllers/concerns/google_authentication.rb [hunk 4]
```diff
@@ -2,7 +2,11 @@ module GoogleAuthentication
   extend ActiveSupport::Concern
 
   def refresh_token
-    session[:credentials] = GoogleOauth.refresh_token(session[:credentials]) if session[:credentials]
+    credentials = session[:credentials]
+    if credentials && GoogleOauth.need_to_refresh_token?(credentials)
```
Comment (quirk0o) [inline app/controllers/concerns/google_authentication.rb:6]: They're always telling us to use `return unless` 😛

**Pull request merged**